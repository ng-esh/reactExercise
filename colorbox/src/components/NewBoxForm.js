import React, { useState } from "react";
import "../styles/NewBoxForm.css";

function NewBoxForm({ addBox }) {
  const [formData, setFormData] = useState({
    width: "",
    height: "",
    backgroundColor: ""
  });

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData(data => ({ ...data, [name]: value }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (!formData.width || !formData.height || !formData.backgroundColor) return;
    
    addBox({ 
      width: Number(formData.width), 
      height: Number(formData.height), 
      backgroundColor: formData.backgroundColor 
    });

    setFormData({ width: "", height: "", backgroundColor: "" });
  };

  return (
    <form className="box-form" onSubmit={handleSubmit}>
      <label htmlFor="width">Width (rem):</label>
      <input 
        type="number" 
        id="width" 
        name="width" 
        value={formData.width} 
        onChange={handleChange} 
        min="1"
      />

      <label htmlFor="height">Height (rem):</label>
      <input 
        type="number" 
        id="height" 
        name="height" 
        value={formData.height} 
        onChange={handleChange} 
        min="1"
      />

      <label htmlFor="backgroundColor">Background Color:</label>
      <input 
        type="text" 
        id="backgroundColor" 
        name="backgroundColor" 
        value={formData.backgroundColor} 
        onChange={handleChange} 
      />

      <button type="submit">Add Box</button>
    </form>
  );
}

export default NewBoxForm;
