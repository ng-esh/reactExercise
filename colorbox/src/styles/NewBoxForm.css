.box-form {
    display: flex;
    flex-direction: column;
    gap: 0.5rem;
    align-items: center;
    margin-bottom: 1rem;
  }
  
  .box-form input {
    padding: 0.5rem;
    font-size: 1rem;
    width: 8rem;
  }
  
  .box-form button {
    background: green;
    color: white;
    border: none;
    padding: 0.5rem;
    font-size: 1rem;
    cursor: pointer;
  }
  
  .box-form button:hover {
    background: darkgreen;
  }
  