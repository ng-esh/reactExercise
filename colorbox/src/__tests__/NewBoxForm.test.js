import { render, fireEvent } from "@testing-library/react";
import NewBoxForm from "../components/NewBoxForm";

test("renders without crashing", () => {
  render(<NewBoxForm addBox={() => {}} />);
});

test("matches snapshot", () => {
  const { asFragment } = render(<NewBoxForm addBox={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

test("calls addBox with correct values", () => {
  const addBox = jest.fn();
  const { getByLabelText, getByText } = render(<NewBoxForm addBox={addBox} />);

  // Updated to match the exact label text
  fireEvent.change(getByLabelText("Width (rem):"), { target: { value: "3" } });
  fireEvent.change(getByLabelText("Height (rem):"), { target: { value: "4" } });
  fireEvent.change(getByLabelText("Background Color:"), { target: { value: "blue" } });

  fireEvent.click(getByText("Add Box"));

  expect(addBox).toHaveBeenCalledWith({ width: 3, height: 4, backgroundColor: "blue" });
});
