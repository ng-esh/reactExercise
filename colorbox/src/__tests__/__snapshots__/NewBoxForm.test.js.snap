// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="box-form"
  >
    <label
      for="width"
    >
      Width (rem):
    </label>
    <input
      id="width"
      min="1"
      name="width"
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Height (rem):
    </label>
    <input
      id="height"
      min="1"
      name="height"
      type="number"
      value=""
    />
    <label
      for="backgroundColor"
    >
      Background Color:
    </label>
    <input
      id="backgroundColor"
      name="backgroundColor"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
